### PR TITLE
Factoring out some Blob helpers.

### PR DESCRIPTION
**Pull request by dhermes**
This is prep work for swapping out the upload implementation to use `google-resumable-media`.

I just wanted to get "unrelated" and non-controversial changes out of the way before a big PR.

### CODE REVIEW

File: storage/google/cloud/storage/blob.py [hunk 1]
```diff
@@ -324,6 +325,23 @@ def delete(self, client=None):
         """
         return self.bucket.delete_blob(self.name, client=client)
 
+    def _make_transport(self, client):
```
Comment (jonparrott) [inline storage/google/cloud/storage/blob.py]: Can this go away when we drop httplib2?

File: storage/google/cloud/storage/blob.py [hunk 2]
```diff
@@ -50,8 +49,10 @@
 
 
 _API_ACCESS_ENDPOINT = 'https://storage.googleapis.com'
+_DEFAULT_CONTENT_TYPE = u'application/octet-stream'
 _DOWNLOAD_URL_TEMPLATE = (
     u'https://www.googleapis.com/download/storage/v1{path}?alt=media')
+_CONTENT_TYPE = 'contentType'
```
Comment (jonparrott) [inline storage/google/cloud/storage/blob.py]: I'd prefer this name be more descriptive, or at least have a comment.

File: storage/google/cloud/storage/blob.py (hunk 1 above)
Comment (dhermes) [inline storage/google/cloud/storage/blob.py]: Yes it can!

File: storage/google/cloud/storage/blob.py (hunk 2 above)
Comment (dhermes) [inline storage/google/cloud/storage/blob.py]: I promise to put it in a follow-up PR. Pinky swear!

**Pull request merged**